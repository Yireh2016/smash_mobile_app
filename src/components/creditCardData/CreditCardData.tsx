/* eslint-disable react-native/no-inline-styles */
import * as React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';
import { Card } from '../../types/Card';
import { H3, P2 } from '../texts/Texts';
import TableRow from '../tableRow/TableRow';
import CreditCardGlass from '../../assets/CreditCardGlass';
import theme from '../../theme/theme';
import CardListItem from '../cardListItem/CardListItem';
import truncateText from '../../helpers/truncateText/truncateText';

interface CreditCardDataProps {
  card: Card;
  onCardPress: (arg0: Card) => void;
  index: number;
}

const styles = StyleSheet.create({
  padding: {
    paddingHorizontal: 12,
    paddingVertical: 18,
  },
});
const CreditCardData: React.FunctionComponent<CreditCardDataProps> = ({
  card,
  onCardPress,
  index,
}) => {
  const { bank, number } = card;

  const onPress = () => {
    onCardPress(card);
  };

  return (
    <>
      <TouchableOpacity testID={`CreditCardData${index}`} onPress={onPress}>
        <CardListItem isLight={true}>
          <View style={styles.padding}>
            <TableRow
              columns={[
                {
                  component: (
                    <View
                      style={{ flexDirection: 'row', alignItems: 'center' }}
                    >
                      <H3>{`${truncateText(bank, 22)}`}</H3>
                      <P2>{` (${number})`}</P2>
                    </View>
                  ),
                  align: 'left',
                  width: 90,
                },
                {
                  component: (
                    <CreditCardGlass
                      color={theme.palette.secondary}
                      factor={1}
                    />
                  ),
                  align: 'right',
                  width: 10,
                },
              ]}
            />
          </View>
        </CardListItem>
      </TouchableOpacity>
    </>
  );
};

export default CreditCardData;
