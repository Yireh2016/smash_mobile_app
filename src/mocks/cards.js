export default [
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '2267',
    current_balance: 500,
    credit_utilization_percentage: 10,
    statement_balance: 2650,
    purchase_apr_percentage: 19,
    min_payment: 65,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 19,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 3100,
      },
    ],
  },
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '9376',
    current_balance: 2000,
    credit_utilization_percentage: 25,
    statement_balance: 1030,
    purchase_apr_percentage: 21,
    min_payment: 37,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 21,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 1100,
      },
    ],
  },
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '9085',
    current_balance: 3000,
    credit_utilization_percentage: 10,
    statement_balance: 1820,
    purchase_apr_percentage: 22,
    min_payment: 42,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 22,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 1900,
      },
    ],
  },
];
