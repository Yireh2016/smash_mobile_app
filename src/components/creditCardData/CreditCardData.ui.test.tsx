import * as React from 'react';
import renderer from 'react-test-renderer';
import CreditCardData from './CreditCardData';
import cards from '../../mocks/cards';
it('CreditCardData renders correctly', () => {
  const ComponentProps = {
    card: cards[0],
    onCardPress: () => {},
    index: 0,
  };
  const Element = () => {
    return <CreditCardData {...ComponentProps} />;
  };

  const rendered = renderer.create(<Element />).toJSON();
  expect(rendered).toMatchSnapshot();
});
