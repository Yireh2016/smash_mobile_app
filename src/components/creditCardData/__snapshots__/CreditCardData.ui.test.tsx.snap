// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCardData renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="CreditCardData0"
>
  <View
    style={
      Object {
        "backgroundColor": "#2C3D49",
        "borderRadius": 12,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 12,
          "paddingVertical": 18,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "width": "90%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 18,
                }
              }
            >
              Citi
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 14,
                }
              }
            >
               (2267)
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "width": "10%",
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={26.888}
            bbWidth={38.073}
            focusable={false}
            height={26.888}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 26,
                  "width": 38,
                },
              ]
            }
            vbHeight={26.888}
            vbWidth={38.073}
            width={38.073}
            xml="


<svg  xmlns=\\"http://www.w3.org/2000/svg\\" 
width=\\"38.073\\"
 height=\\"26.888\\"
viewBox=\\"0 0 38.073 26.888\\">
  <path  d=\\"M112.574,56.462A6.421,6.421,0,1,0,110.6,58.28l4.734,4.707a1.345,1.345,0,1,0,1.948-1.851l-.049-.049Zm-5.243.384a4.141,4.141,0,1,1,2.929-1.216A4.141,4.141,0,0,1,107.331,56.847Z\\" transform=\\"translate(-79.613 -36.552)\\" 
  fill=\\"#29b782\\"/>
  <path  d=\\"M31.45,5.1V3.44A3.44,3.44,0,0,0,28.009,0H3.44A3.44,3.44,0,0,0,0,3.44V5.1a.308.308,0,0,0,.308.308H31.142A.308.308,0,0,0,31.45,5.1Z\\" 
  fill=\\"#29b782\\"/>
  <path  d=\\"M31.142,34.92H.308A.308.308,0,0,0,0,35.228v10.5a3.44,3.44,0,0,0,3.44,3.44H21.977A7.9,7.9,0,0,1,31.45,36.786V35.228a.308.308,0,0,0-.308-.308ZM7.862,43.274a.984.984,0,0,1-.984.984H5.9a.984.984,0,0,1-.984-.984v-.984a.984.984,0,0,1,.984-.981h.981a.981.981,0,0,1,.984.981Z\\" transform=\\"translate(0 -27.549)\\" 
  fill=\\"#29b782\\"/>
</svg>


"
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M112.574,56.462A6.421,6.421,0,1,0,110.6,58.28l4.734,4.707a1.345,1.345,0,1,0,1.948-1.851l-.049-.049Zm-5.243.384a4.141,4.141,0,1,1,2.929-1.216A4.141,4.141,0,0,1,107.331,56.847Z"
                fill={4280924034}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    -79.613,
                    -36.552,
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
              <RNSVGPath
                d="M31.45,5.1V3.44A3.44,3.44,0,0,0,28.009,0H3.44A3.44,3.44,0,0,0,0,3.44V5.1a.308.308,0,0,0,.308.308H31.142A.308.308,0,0,0,31.45,5.1Z"
                fill={4280924034}
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
              <RNSVGPath
                d="M31.142,34.92H.308A.308.308,0,0,0,0,35.228v10.5a3.44,3.44,0,0,0,3.44,3.44H21.977A7.9,7.9,0,0,1,31.45,36.786V35.228a.308.308,0,0,0-.308-.308ZM7.862,43.274a.984.984,0,0,1-.984.984H5.9a.984.984,0,0,1-.984-.984v-.984a.984.984,0,0,1,.984-.981h.981a.981.981,0,0,1,.984.981Z"
                fill={4280924034}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    -27.549,
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</View>
`;
